### PR TITLE
[DO NOT MERGE] WIP/POC: storing in cache and serving it.

### PR DESCRIPTION
**Pull request by frizinak**

### CODE REVIEW

File: src/EventSubscriber/CacheSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            empty($this->expiries)
+            || $response->headers->hasCacheControlDirective('s-maxage')
+        ) {
+            return;
+        }
+
+        $path = $event->getRequest()->getPathInfo();
+        foreach ($this->expiries as $re => $expiry) {
+            // # should be safe... I guess
+            if (!preg_match('#' . $re . '#', $path)) {
```
Comment (RobinHoutevelts) [inline src/EventSubscriber/CacheSubscriber.php]: `if (!$expiry || !preg_match('#' . $re . '#', $path)) {`

Not sure how expensive `preg_match` is, but no need to check match if no expiry is set. <sup>Is that even possible?</sup>

**Comment by frizinak** [comment]
#7

**Comment by frizinak** [comment]
continued in #9

File: src/EventSubscriber/CacheSubscriber.php (hunk 1 above)
Comment (frizinak) [inline src/EventSubscriber/CacheSubscriber.php]: if a regex matches a 0 s-maxage entry we also need to return without setting an actual s-maxage header.

so nah ;)

**Pull request closed without merging**